src/byte_format/deserialize.cr

```diff
@@ -74,13 +74,13 @@ struct Crystalizer::ByteFormat
     Path.new deserialize(String, size)
   end
 
-  # Deserializes a `String` from reading from the `io`, delimited by a trailing `byte_delimiter`.
+  # Deserializes a `String` from reading from the `io`, delimited by a trailing `string_delimiter`.
   def deserialize(to type : String.class)
-    String.build do |str|
-      while (byte = @io.read_byte) && byte != @byte_delimiter
-        str.write_byte byte
-      end
-    end
+    if string_delimiter = @string_delimiter
+      @io.gets string_delimiter, true
```

Comment:
We probably don't want to have `chomp` to true.

Reply:
I think your original implementation did the chomp. You were appending into the string until the delimiter was found, and delimiter wasn't appended.

Reply:
Yes makes sense, I was not sure what was chomped

Reply:
I'd add `chomp: true` to make it more explicit